### PR TITLE
Add Dockerfile for wasm

### PR DESCRIPTION
**Pull request by Meakk**

### CODE REVIEW

File: webassembly/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,52 @@
+FROM dockcross/web-wasm:20240529-0dade71
+
+# Install VTK
+ENV vtk_sha 603cf290299950e64cd3f3463ad3dc3a67b74137
+
+RUN git clone https://github.com/Kitware/VTK.git /vtk-src
```
Comment (mwestphal) [inline webassembly/Dockerfile:6]: gitlab is the true upstream

File: webassembly/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+RUN cd /vtk-src && git checkout $vtk_sha
+
+RUN cmake -S /vtk-src -B /vtk-build \
+  -DBUILD_SHARED_LIBS=OFF \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DCMAKE_INSTALL_PREFIX:PATH=/depends \
```
Comment (mwestphal) [inline webassembly/Dockerfile:13]: not used ?

File: webassembly/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  -DVTK_MODULE_ENABLE_VTK_ImagingCore=YES \
+  -DVTK_MODULE_ENABLE_VTK_ImagingHybrid=YES \
+  -DVTK_MODULE_ENABLE_VTK_InteractionStyle=YES \
+  -DVTK_MODULE_ENABLE_VTK_InteractionWidgets=YES \
+  -DVTK_MODULE_ENABLE_VTK_RenderingAnnotation=YES \
+  -DVTK_MODULE_ENABLE_VTK_RenderingCore=YES \
+  -DVTK_MODULE_ENABLE_VTK_RenderingOpenGL2=YES \
+  -DVTK_MODULE_ENABLE_VTK_RenderingVolumeOpenGL2=YES \
+  -DVTK_MODULE_ENABLE_VTK_TestingCore=YES \
+  -DVTK_SMP_IMPLEMENTATION_TYPE=Sequential \
```
Comment (mwestphal) [inline webassembly/Dockerfile:47]: std::thread not working ?

File: webassembly/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  -DVTK_VERSIONED_INSTALL=OFF
+
+RUN cmake --build /vtk-build --target install
+
+RUN rm -rf /vtk-src /vtk-build
```
Comment (mwestphal) [inline webassembly/Dockerfile:52]: useless as the previous layers still contains the files. you should do it all in a single layer

File: webassembly/Dockerfile (hunk 3 above)
Comment (Meakk) [reply webassembly/Dockerfile:47]: Unfortunately not out of the box. I think it's possible to make it work but I'll take care of it later.

**Review by mwestphal — APPROVED** [review]
Lets add an issue for future improvements though

File: webassembly/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,50 @@
+FROM dockcross/web-wasm:20240529-0dade71
+
+# Install VTK
+ENV vtk_sha 603cf290299950e64cd3f3463ad3dc3a67b74137
```
[the remark this answers is not in the document]
Comment (Meakk) [reply webassembly/Dockerfile:4]: created an issue https://github.com/f3d-app/f3d-docker-images/issues/17

**Pull request merged**